dashboard/app/controllers/maker_controller.rb

```diff
@@ -1,7 +1,17 @@
 class MakerController < ApplicationController
   authorize_resource class: :maker_discount, except: :setup
 
+  # Maker Toolkit is currently used in CSD unit 6, which script id 120.
+  # Retrieves the current CSD unit 6 level that the user is working on.
   def home
+    csd_unit_6_script = Script.find(120)
+    current_level = current_user.next_unpassed_progression_level(csd_unit_6_script)
+    @csd_unit_6 = {
+      assignableName: data_t_suffix('script.name', csd_unit_6_script[:name], 'title'),
+      lessonName: current_level.stage.localized_title,
+      linkToOverview: script_path(csd_unit_6_script),
+      linkToLesson: script_next_path(csd_unit_6_script, 'next')
```

Comment:
Can we generalize/deduplicate this code with the similar code in home_controller.rb?

Reply:
If I did dedupe this little chunk, where should that shared code live?

Reply:
Good question.  Maybe `users_helper.rb` which already contains helpers like `summarize_user_progress` and `level_with_best_progress`?  I'm picturing a signature something like
```rb
def top_course_info(user = current_user, script = user.primary_script)
```
I don't know if you can do dependent defaults like that?